api/src/main/java/org/pmiops/workbench/api/ProfileController.java

```diff
@@ -59,10 +69,92 @@
+          user.getGivenName(), user.getFamilyName());
+    } catch (ApiException e) {
+      log.log(Level.SEVERE, "Error registering user: {0}".format(e.getResponseBody()), e);
+      // We don't expect this to happen.
+      // TODO: figure out what happens if you register after already registering
+      throw new ServerErrorException("Error registering user", e);
+    }
+    WorkbenchConfig workbenchConfig = workbenchConfigProvider.get();
+    long suffix;
+    if (environment.isDevelopment()) {
+      // For local development, make one billing project per account based on a hash of the account
+      // email, and reuse it across database resets. (Assume we won't have any collisions;
+      // if we discover that somebody starts using our namespace, change it up.)
+      suffix = user.getEmail().hashCode();
+    } else {
+      // In other environments, create a suffix based on the user ID from the database. We will
+      // add a suffix if that billing project is already taken. (If the database is reset, we
+      // should consider switching the prefix.)
+      suffix = user.getUserId();
+    }
+    String billingProjectNamePrefix = workbenchConfig.firecloud.billingProjectPrefix + suffix;
+    String billingProjectName = billingProjectNamePrefix;
+    int numAttempts = 0;
+    while (numAttempts < MAX_BILLING_PROJECT_CREATION_ATTEMPTS) {
+      try {
+        fireCloudService.createAllOfUsBillingProject(billingProjectName);
+      } catch (ApiException e) {
+        if (e.getCode() == HttpStatus.CONFLICT.value()) {
+          if (environment.isDevelopment()) {
+            // In local development, just re-use existing projects for the account. (We don't
+            // want to create a new billing project every time the database is reset.)
+            log.log(Level.WARNING, "Project with name {0} already exists; using it."
+                .format(billingProjectName));
+          } else {
+            numAttempts++;
+            // In cloud environments, keep trying billing project names until we find one
+            // that hasn't been used before, or we hit MAX_BILLING_PROJECT_CREATION_ATTEMPTS.
+            billingProjectName = billingProjectNamePrefix + "-" + numAttempts;
+          }
+        } else {
+          log.log(Level.SEVERE, "Error creating billing project: {0}".format(e.getResponseBody()),
+              e);
+          throw new ServerErrorException("Error creating billing project", e);
+        }
+      }
+    }
+    if (numAttempts == MAX_BILLING_PROJECT_CREATION_ATTEMPTS) {
+      throw new ServerErrorException("Encountered {0} billing project name collisions; giving up"
+          .format(String.valueOf(MAX_BILLING_PROJECT_CREATION_ATTEMPTS)));
+    }
+
+    try {
+      fireCloudService.addUserToBillingProject(user.getEmail(), billingProjectName);
+    } catch (ApiException e) {
+      // If we used an existing project above, it's possible this will fail. That should only
```

Comment:
I added special logic for handling 403s here, since that's an error condition that could arise in local development (if somebody other than AofU created a project with the name we're trying to use.) Hopefully that doesn't happen.